document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutor.java

```diff
@@ -16,25 +16,27 @@
 import com.mongodb.client.AggregateIterable;
 import com.mongodb.client.FindIterable;
 import com.mongodb.client.MongoCursor;
-import com.mongodb.client.model.Projections;
 import java.util.List;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 import lombok.AllArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.bson.conversions.Bson;
+import org.hypertrace.core.documentstore.mongo.query.transformer.MongoQueryTransformer;
 import org.hypertrace.core.documentstore.query.Pagination;
 import org.hypertrace.core.documentstore.query.Query;
+import org.hypertrace.core.documentstore.query.QueryInternal;
 
 @Slf4j
 @AllArgsConstructor
 public class MongoQueryExecutor {
   final com.mongodb.client.MongoCollection<BasicDBObject> collection;
 
-  public MongoCursor<BasicDBObject> find(final Query query) {
+  public MongoCursor<BasicDBObject> find(final Query originalQuery) {
+    QueryInternal query = (QueryInternal) originalQuery;
```

Comment:
>If, we are going to cast this, why do we need QueryInternal?
Answered here: https://github.com/hypertrace/document-store/pull/60#discussion_r759220385

> if the originalQuery is not an instance of QueryInternal, would it throw an exception?
As of today, since we do not have any other implementation, this is fine. But, when we have more than one sub-class of Query, the idea is to introduce a factory or so here. However, we anticipate that there won't be a need for any other sub-class of the Query class.